sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobUploadFromUrlOptions.java

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.options;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.storage.blob.models.AccessTier;
+import com.azure.storage.blob.models.BlobHttpHeaders;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.common.implementation.StorageImplUtils;
+
+import java.util.Map;
+
+/**
+ * Extended options that may be passed when uploading a Block Blob from URL.
+ */
+public class BlobUploadFromUrlOptions {
```

Comment:
If this API is on block blob, should the options be BlockBlobUploadFromUrlOptions?

Reply:
@gapra-msft The short term decision is to put this on Block Blob only to let it mature, but long term goal is to put this on BlobClient as well and advertise as first choice over copy.

Reply:
Cool. Sounds good to me then